geonode/resource/handler.py

```diff
@@ -0,0 +1,59 @@
+from abc import ABC
+import logging
+
+logger = logging.getLogger(__file__)
+
+
+class BaseResourceHandler(ABC):
+    """
+    Base abstract resource handler object
+    define the required method needed to define a resource handler
+    As first implementation it will take care of the download url
+    and the download response for a resource
+    """
+
+    REGISTRY = []
+
+    def __init__(self, instance=None) -> None:
+        self.instance = instance
+
+    def __str__(self):
+        return f"{self.__module__}.{self.__class__.__name__}"
+
+    def __repr__(self):
+        return self.__str__()
+
+    @classmethod
+    def register(cls):
```

Comment:
A registry class should be implemented as we do with [AssetHandlerRegistry](https://github.com/GeoNode/geonode/blob/master/geonode/assets/handlers.py#L48), rather then using the base class as a registry itself (which also makes any subclass a registry...).